test/test.py

```diff
@@ -728,6 +728,23 @@ def assert_bundle():
 
         assert_bundle()
 
+    def test_name_substitution(self):
+        uuid = 'ffee7f29-5c38-461a-8771-a68e20ec4a2e'
+        version = '2019-02-02T065454.662896Z'
+        manifest, metadata_files = self._load_bundle(uuid, version, replica='aws', deployment='prod')
+        files_with_bang = [f['name'] for f in manifest if '!' in f['name']]
+        self.assertNotEqual(len(files_with_bang), 0)
+        bundle = Bundle(uuid, version, manifest, metadata_files)
+        entity_json_file_names = set(e.json['file_core']['file_name']
+                                     for e in bundle.entities.values()
+                                     if isinstance(e, AnalysisFile))
+        for file_name in files_with_bang:
```

Comment:
Could be even stronger. This would pass if the CUT replaced bang with something but slash, and the file names with slash were already in the canned bundle. I would write the assertions in terms of set arithmetic.

Reply:
I changed the test to assert the contents verbatim, as per our conversation.